django/db/models/aggregates.py

```diff
@@ -65,7 +65,15 @@ def resolve_expression(
         c.filter = c.filter and c.filter.resolve_expression(
             query, allow_joins, reuse, summarize
         )
-        if not summarize:
+        if summarize:
+            # Summarized aggregates cannot refer to summarized aggregates.
+            for ref in c.get_refs():
+                if query.annotations[ref].is_summary:
+                    raise FieldError(
+                        "Cannot compute %s('%s'): '%s' is an aggregate"
+                        % (c.name, ref, ref)
+                    )
```

Comment:
This restores logic originally in https://github.com/django/django/commit/1297c0d0d76a708017fe196b61a0ab324df76954#diff-d58ef61559dc7af5fdf7b56fee13571a4d2948e784cd608f6afeacf3ac2fb195L598-L608 but in a smarter way as the former not walking expressions recursively like `get_refs` is.